hooks/post_gen_project.py

```diff
@@ -86,3 +87,10 @@ def process_licence(licence_name):
             print(
                 "gitpython is not installed so the repository will not be initialised "
                 "and astropy_helpers not downloaded.")
+
+    else:
+        with urllib.request.urlopen(
```

Comment:
I wonder if it is easier to use [requests](https://2.python-requests.org/en/master/). Then, you can easily check the return code and such. Chances are, more often than we like, people gonna run into timeout when they fall into this logic block. Or recently, there was an issue in core, where they server connection was so unstable that download only succeeded 100% in one out of ten tries or so, resulting in nine other corrupted files. Something to consider.

Reply:
that would be an additional dependency though while urllib is in the standard library.

Reply:
Then maybe find a way to:

1. Catch error from `urlopen` (mainly for timeout).
2. Check that the downloaded file is not corrupted even if `urlopen` seemed successful.

Reply:
how does the previous block behaves in such cases? anyway, we can definitely catch the errors and deal with them.

Reply:
(before I do anything let's wait  for the  others to chime in whether to go with this hack, or add requests as dependency as you first suggested)

Reply:
You could also use [urlretrieve](https://docs.python.org/3.7/library/urllib.request.html#urllib.request.urlretrieve)?

Reply:
yes, I suppose that's a good middle ground solution given that it raises a `ContentTooShortError` when something is fishy with the downloaded file. Thanks @Cadair!